src/data/polynomial/derivative.lean

```diff
@@ -218,6 +218,20 @@ theorem eq_C_of_derivative_eq_zero [no_zero_smul_divisors ℕ R] {f : R[X]} (h :
   f = C (f.coeff 0) :=
 eq_C_of_nat_degree_eq_zero $ nat_degree_eq_zero_of_derivative_eq_zero h
 
+theorem eq_C_iff_eval_and_derivative_eq_zero [no_zero_smul_divisors ℕ R] {c : R} {f : R[X]} :
+f = C c ↔ (∃ r, f.eval r = c) ∧ f.derivative = 0 :=
```

Comment:
Is `no_zero_smul_divisors ℕ R` the same as `char_zero R`?

Reply:
I think no smul zero divisors would be stronger, if you took the product of all Z/n for all n you'd get a ring of char zero with plenty of smul zero divisors from nat.
Whats actually needed here i'm not sure